src/llmcompressor/modifiers/quantization/gptq/base.py

```diff
@@ -175,150 +177,183 @@ def on_initialize_structure(self, state: State, **kwargs):
             self._build_quant_modifier_from_dict(self.quantize)
             self.quantize = True
 
-        if self.quantization_modifier_:
-            self.quantization_modifier_.on_initialize_structure(state, **kwargs)
+        if self._quantization_modifier:
+            self._quantization_modifier.on_initialize_structure(state, **kwargs)
 
     def on_initialize(self, state: "State", **kwargs) -> bool:
         """
         Initialize and run the GPTQ algorithm on the current state
 
         :param state: session state storing input model and calibration data
         """
+        # initialize quantization modifier
         if not self.initialized_structure_:
             self.on_initialize_structure(state, **kwargs)
-        if self.quantization_modifier_:
-            self.quantization_modifier_.initialize(state, **kwargs)
+        if self._quantization_modifier:
+            self._quantization_modifier.initialize(state, **kwargs)
         if not self.quantize:
             raise ValueError("To use the GPTQModifier, quantization must be enabled.")
 
-        modifiable_model = state.model
-        calibration_dataloader = state.data.calib
-
+        # register hooks
+        for name, module in state.model.named_modules():
+            if getattr_chain(module, "quantization_scheme.weights", None) is not None:
+                # HACK: previously, embeddings were not quantized because they were not
+                # accessible by the layer compressor. For now, we manually ignore it,
+                # but in the FUTURE this should be ignored by the user
+                if not isinstance(module, torch.nn.Embedding):
+                    post_hook = partial(self.compress_module, name)
+                    self.register_hook(module, post_hook, "forward")
+
+        # infer sequential targets
         if self.sequential_targets is None:
-            # if no targets are provided, default to the modules that shouldn't be
-            # split by FSDP. For Transformers models this is equivalent to the
-            # decoder layers (ie LlamaDecoderLayer)
-            self.sequential_targets = get_no_split_params(modifiable_model)
-
-        self.initialize_compression(modifiable_model, calibration_dataloader)
-        self.apply_compression(calibration_dataloader)
-        state.model.apply(freeze_module_quantization)
-
-        return True
+            self.sequential_targets = get_no_split_params(state.model)
+        if isinstance(self.sequential_targets, str):
+            self.sequential_targets = [self.sequential_targets]
+
+        # infer update size
+        if self._update_size is None:
+            self._update_size = len(state.data.calib)
+
+        # infer pipeline
+        model_name = state.model.__class__.__name__
+        input_names = state.data.calib.dataset.column_names
+        unfixable_errors = (torch.OutOfMemoryError, torch._C._LinAlgError)
+        try:
+            run_sequential(
```

Comment:
I think we need a more descriptive name to help better differentiate with the layer_sequential pipeline

Reply:
I'm all ears for name suggestions. "sequential" is the term used by LC and AutoGPTQ to refer to the process of breaking a model into parts for the purpose of propagating error / reducing the number of active hessians.

"Layer Sequential Pipeline" and "Sequential Pipeline" were the most cogent names I could think of to describe this process at the time.

Reply:
Could we do "Layer Sequential" and "Subgraph Sequential" ?  Sequential being indicative of the data/error propagation while using "layer" and "subgraph" to differentiate between data structures?

Reply:
I don't really like exposing the idea of subgraphs to the user, as most users won't have the context to really know what that means or why they should choose it.

In the future, the sequential pipeline will be the default for any recipes which include gptq, wanda, or sparsegpt. I'd prefer to keep this name short.

The layer sequential pipeline is only useful as a fallback for a small subset of models which follow layer-wise assumptions (next.input == prev.output) and are not traceable (due to weird or bad programming). The only model I've found like this is minicpm.

Reply:
hm let me think of other descriptors 
I think we just want each of the pipelines beyond the basic pipeline to be a little more verbose in its name

Reply:
We do expose the idea of "tracing". Just something to think about, I'll sleep on it too